src/BreweryFactory.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Interns2022B;
+
+use Illuminate\Support\Collection;
+use Interns2022B\Models\Brewery;
+use Interns2022B\Models\City;
+use Interns2022B\Models\Country;
+use Interns2022B\Models\Provider;
+
+class BreweryFactory
+{
+    public function create(array $breweryData): Collection
+    {
+        $transformed = new Collection();
+        foreach ($breweryData as $row) {
+            $brewery = new Brewery(
+                name: $row["name"],
+                city: new City(name: $row["city"]),
+                provider: new Provider(
+                    name: $row["provider"],
+                    breweries: new Collection((array)$row["name"]),
```

Comment:
We should not create for every brewery new provider because one provider can have many breweries. I think, that problem will be good for separate issue.

Reply:
changed